<!--
SPDX-FileCopyrightText: 2026 Alexandre Gomes Gaigalas <alganet@gmail.com>

SPDX-License-Identifier: ISC
-->

# Age Header Numeric

## Description

The `Age` header, when present in a response, indicates the time (in seconds) since the response was generated by an origin server. This rule ensures the `Age` header value is a non-negative integer (`delta-seconds`) as required by the HTTP specifications.

## Specifications

- [RFC 9111 §5.1](https://www.rfc-editor.org/rfc/rfc9111.html#section-5.1) — `Age` field-value: delta-seconds (non-negative integer).

## Configuration

```toml
[rules.message_age_header_numeric]
enabled = true
severity = "warn"
```

## Examples

### ✅ Good

```http
HTTP/1.1 200 OK
Age: 120
```

### ❌ Bad

```http
HTTP/1.1 200 OK
Age: -1
```

```http
HTTP/1.1 200 OK
Age: abc
```
